### PR TITLE
Fix system rspec tests

### PR DESCRIPTION
**Pull request by britaumich**
All spec/system tests should run.
Commands:

- SHOW_BROWSER=true rspec spec/system/<file_name>
- rspec spec/system/<file_name>
- rspec spec/system/*

### CODE REVIEW

**Comment by britaumich** [comment]
@Mohammad4844  
I edited rspec tests moving mock_login(user) after allow(LdapLookup)... statements.
Now they work with no network.
That was a great suggestion.
I removed all sleep commands to run tests faster.

**Review by Mohammad4844 — CHANGES_REQUESTED** [review]
Some of the tests fail **_sometimes_**. They throw a Selenium error, and I'm pretty sure it's because the rspec code runs faster than the server can respond (it only happens when you're dealing with alert popups, like when it asks 'are you sure you want to delete this?'). 

I tried using capybara's built in helpers to deal with alerts (`dismiss_confirm` and `accept_confirm`), and they seem to solve the issue since they have a built in waiting functionality. I added the code as comments to each of the test case to make it work.

File: spec/system/common_attributes_controller_spec.rb [hunk 1]
```diff
@@ -51,41 +56,32 @@
...
 
     it 'click on delete icon and cancel the alert messege' do
       VCR.use_cassette "common_attribute" do
         visit common_attributes_path
-        sleep 2
         find(:css, 'i.bi.bi-trash-fill.text-danger').click
-        sleep 2
         text = page.driver.browser.switch_to.alert.text
         expect(text).to eq 'Are you sure you want to delete this Common Attribute?'
         page.driver.browser.switch_to.alert.dismiss
```
Comment (Mohammad4844) [inline spec/system/common_attributes_controller_spec.rb:73]:
```
dismiss_confirm 'Are you sure you want to delete this Common Attribute?' do
  find(:css, 'i.bi.bi-trash-fill.text-danger').click
end
```

File: spec/system/zones_controller_spec.rb [hunk 2]
```diff
@@ -51,41 +45,37 @@
...
       VCR.use_cassette "zone" do
-        visit zones_path
         sleep 2
+        visit zones_path
         find(:css, 'i.bi.bi-trash-fill.text-danger').click
-        sleep 2
         text = page.driver.browser.switch_to.alert.text
+        sleep 2
         expect(text).to eq 'Are you sure you want to delete this zone?'
         page.driver.browser.switch_to.alert.dismiss
```
Comment (Mohammad4844) [inline spec/system/zones_controller_spec.rb:64]:
```
dismiss_confirm 'Are you sure you want to delete this zone?' do
  find(:css, 'i.bi.bi-trash-fill.text-danger').click
end
```

File: spec/system/zones_controller_spec.rb [hunk 3]
```diff
@@ -51,41 +45,37 @@
...
       VCR.use_cassette "zone" do
-        visit zones_path
         sleep 2
+        visit zones_path
         find(:css, 'i.bi.bi-trash-fill.text-danger').click
-        sleep 2
         text = page.driver.browser.switch_to.alert.text
+        sleep 2
         expect(text).to eq 'Are you sure you want to delete this zone?'
         page.driver.browser.switch_to.alert.accept
```
Comment (Mohammad4844) [inline spec/system/zones_controller_spec.rb:78]:
```
accept_confirm 'Are you sure you want to delete this zone?' do
  find(:css, 'i.bi.bi-trash-fill.text-danger').click
end
```

**Comment by britaumich** [comment]
@Mohammad4844 All tests right now run as single files and all together.
Please check it they run for you.
I'll keep dismiss_browser_dialog and accept_browser_dialog just for reference.

**Review by palinaskakun — APPROVED** [review]
code looks good, all tests pass, no errors

**Review by charlieezh — DISMISSED** [review]
all tests passed on my end, looking good

**Review by Mohammad4844 — APPROVED** [review]

**Pull request merged**